spring-integration-websocket/src/main/java/org/springframework/integration/websocket/inbound/WebSocketInboundChannelAdapter.java

```diff
@@ -234,13 +236,23 @@ public void afterSessionStarted(WebSocketSession session) {
 				SubProtocolHandler protocolHandler = this.subProtocolHandlerRegistry.findProtocolHandler(session);
 				protocolHandler.afterSessionStarted(session, this.subProtocolHandlerChannel);
 				if (!this.server && protocolHandler instanceof StompSubProtocolHandler) {
+					// The CONNECT frame is required by the STOMP specification.
 					StompHeaderAccessor accessor = StompHeaderAccessor.create(StompCommand.CONNECT);
 					accessor.setSessionId(session.getId());
 					accessor.setLeaveMutable(true);
 					accessor.setAcceptVersion("1.1,1.2");
 
-					Message<?> connectMessage =
+					Message<byte[]> connectMessage =
 							MessageBuilder.createMessage(EMPTY_PAYLOAD, accessor.getMessageHeaders());
+
+					// In the client mode, the client session has to register itself
+					// into the StompSubProtocolHandler cache
+					// for proper correlation of the messages from the server side.
+					StompEncoder stompEncoder = new StompEncoder();
```

Comment:
Could the StompEncoder be created as a class attribute and initialized in the constructor?